core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -58,15 +61,43 @@ public DockerComposeContainer(File composeFile, String command) {
 
     @Override
     public void start() {
+
+        final Profiler profiler = new Profiler("Docker compose container rule");
+        profiler.setLogger(logger());
+        profiler.start("Docker compose container startup");
+
+        // Start the docker-compose container, which starts up the services
+        super.start();
+        followOutput(new Slf4jLogConsumer(logger()), OutputFrame.OutputType.STDERR);
+
+        // wait for the compose container to stop, which should only happen after it has spawned all the service containers
+        logger().info("Docker compose container is running - service creation will start now");
+        while (isRunning()) {
+            logger().trace("Compose container is still running");
+            Uninterruptibles.sleepUninterruptibly(100, TimeUnit.MILLISECONDS);
+        }
+        logger().info("Docker compose has finished running");
+
+        // Ensure that all service containers that were launched by compose will be killed at shutdown
+        try {
+            List<Container> containers = dockerClient.listContainersCmd().withShowAll(true).exec();
```

Comment:
`inspectCmd` would be better. Also, even if something went wrong, we still should call `ContainerReaper`. Currently, it will be called only if name (which used for logging only) was found